Problems with unsigned char on PowerPC

Some places of the code assume that char is signed per default.  That’s not the case on powerpc, which leads to the following warnings (breaking the build with -Werror):

```
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp: In member function 'vi
rtual bool OpenImageIO::v1_4::CineonInput::open(const std::string&, OpenImageIO:
:v1_4::ImageSpec&)':
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:298: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:314: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:318: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:319: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:320: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:321: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:327: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:328: warning: compariso
n is always true due to limited range of data type
/home/pascal/src/oiio/src/cineon.imageio/cineoninput.cpp:329: warning: compariso
n is always true due to limited range of data type

[...]

/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp: In member function 'virtual 
bool OpenImageIO::v1_4::DPXInput::seek_subimage(int, int, OpenImageIO::v1_4::Ima
geSpec&)':
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:382: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:429: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:430: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:431: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:432: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:433: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:434: warning: comparison is a
lways true due to limited range of data type
/home/pascal/src/oiio/src/dpx.imageio/dpxinput.cpp:435: warning: comparison is a
lways true due to limited range of data type

[...]

/home/pascal/src/oiio/src/rla.imageio/rlainput.cpp: In member function 'size_t O
penImageIO::v1_4::RLAInput::decode_rle_span(unsigned char*, int, int, const char
*, size_t)':
/home/pascal/src/oiio/src/rla.imageio/rlainput.cpp:453: warning: comparison is a
lways true due to limited range of data type

```

I don’t know a lot about the code, but it looks like this is a genuine problem that should not be ignored.

--- a/src/cineon.imageio/cineoninput.cpp
+++ b/src/cineon.imageio/cineoninput.cpp
@@ -271,7 +271,7 @@ CineonInput::open (const std::string &name, ImageSpec &newspec)
                                                 TypeDesc (TypeDesc::FLOAT,  \
                                                 2), &floats[0])
 #define CINEON_SET_ATTRIB_STR(X, x)     if (m_cin.header.x[0]               \
-                                        && m_cin.header.x[0] != -1)         \
+                                        && m_cin.header.x[0] != char(-1))   \
                                             m_spec.attribute ("cineon:" #X, \
                                                 m_cin.header.x)
 

--- a/src/dpx.imageio/dpxinput.cpp
+++ b/src/dpx.imageio/dpxinput.cpp
@@ -379,7 +379,7 @@ DPXInput::seek_subimage (int subimage, int miplevel, ImageSpec &newspec)
         m_spec.attribute ("compression", "rle");
     char buf[32 + 1];
     m_dpx.header.Description (subimage, buf);
-    if (buf[0] && buf[0] != -1)
+    if (buf[0] && buf[0] != char(-1))
         m_spec.attribute ("ImageDescription", buf);
     m_spec.attribute ("PixelAspectRatio", m_dpx.header.AspectRatio(0)
          / (float)m_dpx.header.AspectRatio(1));
@@ -407,7 +407,7 @@ DPXInput::seek_subimage (int subimage, int miplevel, ImageSpec &newspec)
                                         DPX_SET_ATTRIB(x, )
     // see comment above Copyright, Software and DocumentName
 #define DPX_SET_ATTRIB_STR(X, x)    if (m_dpx.header.x[0]                     \
-                                        && m_dpx.header.x[0] != -1)           \
+                                        && m_dpx.header.x[0] != char(-1))     \
                                         m_spec.attribute ("dpx:" #X,          \
                                             m_dpx.header.x)
 

--- a/src/rla.imageio/rlainput.cpp
+++ b/src/rla.imageio/rlainput.cpp
@@ -449,7 +449,7 @@ RLAInput::decode_rle_span (unsigned char *buf, int n, int stride,
 {
     size_t e = 0;
     while (n > 0 && e < elen) {
-        char count = encoded[e++];
+        signed char count = (signed char) encoded[e++];
         if (count >= 0) {
             // run count positive: value repeated count+1 times
             for (int i = 0;  i <= count && n;  ++i, buf += stride, --n)

